app/loader.cpp

```diff
@@ -85,3 +99,10 @@ QStringList Loader::mapTip(QgsQuickFeatureLayerPair pair)
     }
     return previewFields;
 }
+
+void Loader::appStateChanged(Qt::ApplicationState state)
+{
+    if (!mIsRecordingOn) {
+        mPositionKit->requestUpdates(state == Qt::ApplicationActive);
+    }
```

Comment:
can you please add a comment here that this is meant to save battery by turning GPS off when the app is the background?